File: WinForms/VB/DiagramMainDemo/Modules/ImagesModule.vb
Namespace DevExpress.XtraDiagram.Demos

    Public Partial Class ImagesModule
        Inherits DocumentModuleBase

        Public Sub New()
            InitializeComponent()
            AddHandler diagramControl.MouseDoubleClick, New System.Windows.Forms.MouseEventHandler(AddressOf DiagramControl_MouseClick) 'VB
        End Sub

        Protected Overrides ReadOnly Property DocumentName As String
            Get
                Return "Images.xml"
            End Get
        End Property

        Private Sub DiagramControl_MouseClick(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs)
            If e.Button <> System.Windows.Forms.MouseButtons.Left Then Return
            Dim imageItem = TryCast(diagramControl.CalcHitItem(New System.Drawing.PointF(e.X, e.Y)), DiagramImage)
            If imageItem Is Nothing OrElse Not Equals(imageItem.Tag, GetType(DiagramImage).Name) Then Return
            diagramControl.SelectItem(imageItem)
            diagramControl.LoadImage()
        End Sub
    End Class
End Namespace

File: WinForms/CS/DiagramMainDemo/Modules/ImagesModule.cs
namespace DevExpress.XtraDiagram.Demos {
    public partial class ImagesModule : DocumentModuleBase {
        public ImagesModule() {
            InitializeComponent();
            this.diagramControl.MouseDoubleClick += new System.Windows.Forms.MouseEventHandler(this.DiagramControl_MouseClick);//VB
        }
        protected override string DocumentName { get { return "Images.xml"; } }

        void DiagramControl_MouseClick(object sender, System.Windows.Forms.MouseEventArgs e) {
            if(e.Button != System.Windows.Forms.MouseButtons.Left)
                return;
            var imageItem = diagramControl.CalcHitItem(new System.Drawing.PointF(e.X, e.Y)) as DiagramImage;
            if(imageItem == null || !object.Equals(imageItem.Tag, typeof(DiagramImage).Name))
                return;
            diagramControl.SelectItem(imageItem);
            diagramControl.LoadImage();
        }
    }
}
